### PR TITLE
Feature #184

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

**Comment by je-ik** [comment]
Currently, there is no "magical" logic that could be used to split partitions at runtime. It only pre-splits partitions and then processes them one-by-one. I think we can adopt a runtime-splittable approach when (and if) this is supported by Flink, or we get use-cases for Dataflow.

**Comment by je-ik** [comment]
This closes O2-Czech-Republic#184

**Review by dmvk — APPROVED** [review]
LGTM 👍

File: beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java [hunk 1]
```diff
@@ -280,8 +280,7 @@ public void reload() {
             .flatMap(d -> d.getAttributes().stream())
             .distinct()
             .collect(Collectors.toList());
-    AttributeDescriptor<?>[] closureAsArray =
-        attrClosure.toArray(new AttributeDescriptor[attrClosure.size()]);
+    AttributeDescriptor<?>[] closureAsArray = attrClosure.toArray(new AttributeDescriptor[0]);
```
[the remark this answers is not in the document]
Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/core/BeamDataOperator.java:283]: https://en.wikipedia.org/wiki/Transitive_closure

**Pull request merged**